x-pack/solutions/observability/plugins/infra/server/features.ts

```diff
@@ -34,7 +34,7 @@ const metricRuleTypes = [
 
 const metricAlertingFeatures = metricRuleTypes.map((ruleTypeId) => ({
```

Comment:
The custom threshold rule is available to users with `logs`, `infra`, or `stackAlerts` Kibana privilege. However, the producer remains `observability`. 

Although the producer is now set to a non-existing feature id, this doesn't appear to be an issue, even when the consumer is 'alerts', because the fallback to `producer` logic no longer seems to be in place. Rather, this has been superseded by `authorizedConsumer` logic in [this PR](https://github.com/elastic/kibana/pull/183756). @cnasikas could you confirm and help explain this change? 
![image](https://github.com/user-attachments/assets/57568905-fec4-494a-9d06-fa10acc2a29d)

Likewise, in some cases the `consumer` may also be `observability` which is why we added the `observability` as a valid consumer for all rules.